pkg/interface/chat/src/js/components/member.js

```diff
@@ -5,130 +5,117 @@ import urbitOb from 'urbit-ob';
+          api={props.api}  />
+      );
+    });
 
-    let peers = props.peers[state.station] || [window.ship];
-    let listMembers = peers.map((mem) => {
+    let readListMembers = readGroup.map((mem) => {
       return (
         <MemberElement 
           key={mem} 
-          host={state.host}
+          owner={deSig(props.match.params.ship)}
           ship={mem}
-          circle={state.circle}
+          path={`/chat${state.station}/read`}
+          kind={props.read.kind}
           api={props.api} />
       );
     });
 
-    let errorElem = !!this.state.error ? (
-      <p className="pa2 nice-red label-regular">Invalid ship name.</p>
-    ) : (
-      <div></div>
-    );
-
-    let successElem = !!this.state.success ? (
-      <p className="pa2 nice-green label-regular">Sent invites!</p>
-    ) : (
-      <div></div>
-    );
-
-
-    let inviteButtonClasses = "label-regular black underline btn-font pointer";
-    if (!this.state.error) {
-      inviteButtonClasses = inviteButtonClasses + ' black';
-    }
-
     return (
       <div className="h-100 w-100 overflow-x-hidden flex flex-column">
         <div className='pl3 pt2 bb mb3'>
-          <h2>{state.circle}</h2>
+          <h2>{state.station.substr(1)}</h2>
           <ChatTabBar
             {...props}
             station={state.station}
-            numPeers={peers.length} />
+            numPeers={writeGroup.length}
+            isOwner={deSig(props.match.params.ship) === window.ship} />
         </div>
         <div className="w-100 cf">
-          <div className="w-50 fl pa2">
-            <p className="body-regular">Permitted Members</p>
+          <div className="w-50 fl pa2 pr3">
+            <p className="body-regular mb3">Members</p>
+            <p className="label-regular gray mb3">{writeText}</p>
+            {writeListMembers}
+          </div>
+          <div className="w-50 fr pa2 pl3">
+            <p className="body-regular mb3">Modify Permissions</p>
+            <p className="label-regular gray mb3">
+              {modWriteText}
+            </p>
+            { window.ship === deSig(props.match.params.ship) ? (
+              <InviteElement
+                path={`/chat${state.station}/write`}
+                station={state.station}
+                permissions={props.write}
+                api={props.api} />
+            ) : null }
+          </div>
+        </div>
+        <div className="w-100 cf mt2">
+          <div className="w-50 fl pa2 pr3">
+            <p className="label-regular gray mb3">{readText}</p>
+            {readListMembers}
+          </div>
+          <div className="w-50 fr pa2 pl3">
             <p className="label-regular gray mb3">
-              Everyone with permission to see this chat.
+              {modReadText}
             </p>
-            {listMembers}
+            { window.ship === deSig(props.match.params.ship) ?
+              (  <InviteElement
+                   path={`/chat${state.station}/read`}
```

Comment:
I've noticed that some paths are `/chat/room/station` while others are `/chat/station`.  This is probably intended, but it's worth verifying things behave as you expect.

Reply:
Yeah, after seeing more of this code I'm guessing it's intended.  But nonetheless, worth another look.